addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -106,10 +106,25 @@ def _timesheet_prepare_line_values(self, index, work_hours_data, day_date, work_
             'company_id': self.holiday_status_id.timesheet_task_id.company_id.id or self.holiday_status_id.timesheet_project_id.company_id.id,
         }
 
+    def _check_missing_global_leave_timesheets(self):
+        min_date = min([leave.date_from for leave in self])
+        max_date = max([leave.date_to for leave in self])
+
+        global_leaves = self.env['resource.calendar.leaves'].search([
+            ("resource_id", "=", False),
+            ("date_to", ">=", min_date),
+            ("date_from", "<=", max_date),
+            ("calendar_id.company_id.internal_project_id", "!=", False),
+            ("calendar_id.company_id.leave_timesheet_task_id", "!=", False),
```

Comment:
Why not using the company_id field defined in `resource.calendar.leave` model ? 🧐 
```suggestion
            ("company_id.internal_project_id", "!=", False),
            ("company_id.leave_timesheet_task_id", "!=", False),
```

Reply:
Yeah I'm also scratching my head trying to figure out why I wrote it like this 😅